Fetch cards data from API and handle errors.

cianoid opened the pull request.

cianoid committed:
Fetch cards data from API and handle errors.

Replaced hardcoded card data with dynamic API fetch. Added error handling logic to manage HTTP and custom errors with a mapped error message system. Cards now load dynamically on component mount.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -8,41 +8,54 @@ function getTurned() {
+        errorCode: cards.value.error.code,
+        errorText: errorMap.has(cards.value.error.code) ? errorMap.get(cards.value.error.code) : cards.value.error.message,
+      };
+    }
+  } else {
```
Эти строки кода (data[i].state = "closed"; и data[i].status = "pending";) находятся вне какой-либо функции или цикла, что приведет к ошибке выполнения. Если вы хотите добавить статусы к карточкам, это нужно делать после получения данных, например, внутри функции loadCards, итерируя по массиву cards.value.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -8,41 +8,54 @@ function getTurned() {
+  } else {
+    error.value = {};
+  }
+}
```
Строка <div v-if="cards" class="cards"> находится вне шаблона компонента Vue. В SFC (Single File Component) шаблон должен быть внутри блока <template>. Необходимо перенести разметку внутрь <template> и корректно реализовать вывод карточек.

cianoid committed:
Refactor card initialization and simplify computed property.

Moved card initialization logic into the main function to remove redundancy and streamline data handling. Removed the unnecessary `cardsData` computed property and updated the `Card` component to use `cards` directly.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -8,41 +8,47 @@ function getTurned() {
+  } else {
+    error.value = {};
+
+    for (let i = 0; i < cards.value.length; i++) {
+      cards.value[i].number = i > 9 ? `${i + 1}` : `0${i + 1}`;
+      cards.value[i].state = "closed";
```
Разметка и стили находятся вне блока <template> и <style>. В компонентах Vue разметка должна быть внутри <template>, а стили — внутри <style>. Необходимо перенести эти строки в соответствующие блоки. Также убедитесь, что компонент Card импортирован и определён, а метод getTurned реализован в вашем компоненте.
